src/model/document.js

```diff
@@ -391,6 +394,10 @@ export default class Document {
 
 		do {
 			for ( const callback of this._postFixers ) {
+				// Ensure selection attributes are up to date before each post-fixer.
+				// https://github.com/ckeditor/ckeditor5-engine/issues/1673.
```

Comment:
I'd add here additional comment on why it is done like this and not after every operation so we won't be re-thinking this problem again in a year.